modules/library/main/src/main/java/org/geotools/filter/function/InterpolateFunction.java

```diff
@@ -378,6 +372,32 @@ public <T> T evaluate(Object object, Class<T> context) {
         }
     }
 
+    @SuppressWarnings("unchecked")
+    private <T> Class<T> sanitizeContext(Class<T> context) {
+        if (null == context || Object.class.equals(context)) {
+            if (method == Method.NUMERIC) {
+                context = (Class<T>) Double.class;
+            } else if (method == Method.COLOR) {
+                context = (Class<T>) Color.class;
+            } else {
+                throw new IllegalStateException("Unknown method, expected NUMERIC or COLOR");
```

Comment:
Can you double-check the Symbology Encoding 1.1 specification and see if there is a default? 
I have vague memories it should be numeric, and the "sample" value for the parameter is indeed that, but I'm not sure.

Reply:
The default of mode LINEAR and method NUMERIC are set in `initialize()`, hence the IAE here, it should never be reached for as long as `initialize()` is called before

Reply:
note both cases are fully covered by tests:
![image](https://github.com/geotools/geotools/assets/207423/8bd16435-3a2a-4f56-9cf0-bdca60f328c9)